src/main/java/core/basesyntax/service/RegistrationServiceImpl.java

```diff
@@ -1,30 +1,39 @@
 package core.basesyntax.service;
 
 import core.basesyntax.dao.StorageDaoImpl;
-import core.basesyntax.db.Storage;
 import core.basesyntax.model.User;
 
 public class RegistrationServiceImpl implements RegistrationService {
+    private StorageDaoImpl dao = new StorageDaoImpl();
 
     @Override
     public User register(User user) {
+        checkUserDataForTrueValues(user);
+        if (user.getAge() < 18) {
+            throw new RuntimeException("User is not adult!");
+        }
+        if (dao.get(user.getLogin()) != null) {
+            throw new RuntimeException("Login is exist!");
+        }
+        return dao.add(user);
+    }
+
+    private void checkUserDataForTrueValues(User user) {
         if (user == null) {
             throw new RuntimeException("Incorrect entry about user");
         }
-        if (user.getPassword() == null || user.getLogin() == null) {
-            throw new RuntimeException("User's password or login is empty");
+        if (user.getLogin() == null || user.getPassword() == null) {
+            throw new RuntimeException("User's login or password is absent");
         }
-        if ( user.getAge() < 18) {
-            throw new RuntimeException(" User is not adult! Registration canceled.");
+        if (user.getPassword().length() < 6) {
+            throw new RuntimeException("User's password is very short");
         }
-        if ( user.getPassword().length() < 6) {
-            throw new RuntimeException("User's password is very short. Registration canceled");
+        Integer age = user.getAge();
+        if (age == null) {
+            throw new RuntimeException("User's age is empty");
         }
-        StorageDaoImpl dao = new StorageDaoImpl();
-        if (dao.get(user.getLogin()) != null) {
-            throw new RuntimeException("Login is exist");
+        if (age < 0 || age > 130) {
+            throw new RuntimeException("Incorrect user's age. Age =" + user.getAge());
```

Comment:
there is nothing in the checklist about age ranges, only if less 18 or not

Reply:
done